electron/main/electron-store/ipcHandlers.ts

```diff
@@ -191,9 +199,68 @@ export const registerStoreHandlers = (store: Store<StoreSchema>, windowsManager:
     const chatHistoriesMap = store.get(StoreKeys.ChatHistories)
     const allChatHistories = chatHistoriesMap[vaultDir] || []
     const filteredChatHistories = allChatHistories.filter((item) => item.id !== chatID)
+
     chatHistoriesMap[vaultDir] = filteredChatHistories.reverse()
     store.set(StoreKeys.ChatHistories, chatHistoriesMap)
   })
+
+  ipcMain.handle('get-current-open-files', () => store.get(StoreKeys.OpenTabs) || [])
+
+  ipcMain.handle('set-current-open-files', (event, action, args) => {
+    const openTabs: Tab[] = store.get(StoreKeys.OpenTabs) || []
+
+    const addTab = ({ tab }: { tab: Tab }) => {
+      if (tab === null) return
+      const existingTab = openTabs.findIndex((item) => item.filePath === tab.filePath)
+
+      /* If tab is already open, do not do anything */
+      if (existingTab !== -1) return
+
+      openTabs.push(tab)
+      store.set(StoreKeys.OpenTabs, openTabs)
+    }
+
+    const removeTab = ({ tabId, idx, newIndex }: { tabId: string; idx: number; newIndex: number }) => {
+      openTabs[idx].lastAccessed = false
+      openTabs[newIndex].lastAccessed = true
+      const updatedTabs = openTabs.filter((tab) => tab.id !== tabId)
+      store.set(StoreKeys.OpenTabs, updatedTabs)
```

Comment:
Yeah, I'll go ahead and add it, but either way I'm pretty sure it is not possible for `idx `and `newIndex `not to be valid since tabId is guaranteed to exist which means `idx `and `newIndex `must exist.